datafusion/core/src/physical_optimizer/aggregate_statistics.rs

```diff
@@ -293,38 +297,80 @@ mod tests {
+            true => (Field::new("COUNT(a)", DataType::Int64, false), 2),
         };
 
         // A ProjectionExec is a sign that the count optimization was applied
         assert!(optimized.as_any().is::<ProjectionExec>());
+        let task_ctx = session_ctx.task_ctx();
         let result = common::collect(optimized.execute(0, task_ctx)?).await?;
         assert_eq!(result[0].schema(), Arc::new(Schema::new(vec![col])));
         assert_eq!(
             result[0]
                 .column(0)
                 .as_any()
-                .downcast_ref::<UInt64Array>()
+                .downcast_ref::<Int64Array>()
                 .unwrap()
                 .values(),
             &[count]
         );
+
+        // Validate that the optimized plan returns the exact same
+        // answer (both schema and data) as the original plan
+        let task_ctx = session_ctx.task_ctx();
+        let plan_result = common::collect(plan.execute(0, task_ctx)?).await?;
+        assert_eq!(normalize(result), normalize(plan_result));
         Ok(())
     }
 
+    /// Normalize record batches for comparison:
+    /// 1. Sets nullable to `true`
+    fn normalize(batches: Vec<RecordBatch>) -> Vec<RecordBatch> {
+        let schema = normalize_schema(&batches[0].schema());
+        batches
+            .into_iter()
+            .map(|batch| {
+                RecordBatch::try_new(schema.clone(), batch.columns().to_vec())
+                    .expect("Error creating record batch")
+            })
+            .collect()
+    }
+    fn normalize_schema(schema: &Schema) -> Arc<Schema> {
+        let nullable = true;
+        let normalized_fields = schema
+            .fields()
+            .iter()
+            .map(|f| {
+                Field::new(f.name(), f.data_type().clone(), nullable)
+                    .with_metadata(f.metadata().cloned())
+            })
+            .collect();
+        Arc::new(Schema::new_with_metadata(
+            normalized_fields,
+            schema.metadata().clone(),
+        ))
+    }
+
     fn count_expr(schema: Option<&Schema>, col: Option<&str>) -> Arc<dyn AggregateExpr> {
-        // Return appropriate expr depending if COUNT is for col or table
-        let expr = match schema {
-            None => expressions::lit(ScalarValue::UInt8(Some(1))),
-            Some(s) => expressions::col(col.unwrap(), s).unwrap(),
+        // Return appropriate expr depending if COUNT is for col or table (*)
+        let (expr, name) = match schema {
+            None => (
+                expressions::lit(COUNT_STAR_EXPANSION),
+                COUNT_STAR_NAME.to_string(),
+            ),
+            Some(s) => (
+                expressions::col(col.unwrap(), s).unwrap(),
+                format!("COUNT({})", col.unwrap()),
+            ),
         };
-        Arc::new(Count::new(expr, "my_count_alias", DataType::UInt64))
+
+        Arc::new(Count::new(expr, name, DataType::Int64))
```

Comment:
Now that the schema is checked, we can't use some arbitrary column name, we need to use the actual name the plan would